### PR TITLE
sof-kernel-log-check.sh: change sed -n $begin,\$p to tail -n +$begin 

### PR DESCRIPTION
**Pull request by marc-hb**
This allows removing "eval" and some quoting headaches.

Also fix the last shellcheck warnings

### CODE REVIEW

**Review by xiulipan — CHANGES_REQUESTED** [review]
I think `declare -p` will output some debug logs. You can keep it as comment out to help developer to debug later but I do not think we should enable it by default.

```
2020-10-14 06:19:16 UTC [REMOTE_INFO] Check sof debug fs environment
2020-10-14 06:19:16 UTC [REMOTE_INFO] Checking ipc flood test!
2020-10-14 06:19:16 UTC [REMOTE_INFO] ===== [1/1] loop Begin =====
2020-10-14 06:19:16 UTC [REMOTE_COMMAND] sudo bash -c 'echo 10000 > /sys/kernel/debug/sof/ipc_flood_count'
declare -- cmd="dmesg"
2020-10-14 06:19:17 UTC [REMOTE_INFO] Dumping test logs!
[  101.921335] sof-audio-pci 0000:00:1f.3: IPC Flood count: 10000, Avg response time: 53016ns
[  101.921335] sof-audio-pci 0000:00:1f.3: Max response time: 123315ns
[  101.921336] sof-audio-pci 0000:00:1f.3: Min response time: 36133ns
declare -- cmd="tail -n  +1283  /var/log/kern.log"
2020-10-14 06:19:17 UTC [REMOTE_INFO] Test Result: PASS!
```

File: tools/sof-kernel-log-check.sh [hunk 1]
```diff
@@ -111,26 +111,25 @@ esac
...
+#[[ $kernel_buffer_size -lt $dmesg ]] && cmd="dmesg" || cmd="tail -n  +${begin_line}  /var/log/kern.log"
 
 # confirm begin_line is number, if it is not the number, direct using dmesg
 [[ "${begin_line//[0-9]/}" ]] && begin_line=0
-[[ "$begin_line" -eq 0 ]] && cmd="dmesg" || cmd="sed -n '$begin_line,\$p' /var/log/kern.log"
+[[ "$begin_line" -eq 0 ]] && cmd="dmesg" || cmd="tail -n  +${begin_line}  /var/log/kern.log"
 
 #echo "run $0 with parameter '$*' for check kernel message error"
 
+declare -p cmd
```
[the remark this answers is not in the document]
Comment (marc-hb) [reply tools/sof-kernel-log-check.sh:122]: No, just useful logging.

[the remark this answers is not in the document]
Comment (marc-hb) [reply tools/sof-kernel-log-check.sh:122]: This log just one line and for now this code is a complicated mess. In commit 8b1092e7b46f you added many more lines to debug an issue in the same area.

[the remark this answers is not in the document]
Comment (marc-hb) [reply tools/sof-kernel-log-check.sh]: I commented it out because I care about all the rest much more.

**Review by xiulipan — APPROVED** [review]
@marc-hb Thanks for the clean up. I think the debug log would be useful when we find issues. We should not enable it by default.

**Comment by marc-hb** [comment]
>  I think the debug log would be useful when we find issues. We should not enable it by default.

I agree except we have _already_ found issues! see #366, your commit 8b1092e7b4, https://github.com/thesofproject/sof-test/pull/354#discussion_r482769591 and probably others

**Pull request merged**